verify/update-bazel.sh

```diff
@@ -20,31 +20,22 @@ set -o pipefail
 REPOINFRA_ROOT=$(git rev-parse --show-toplevel)
 # https://github.com/kubernetes/test-infra/issues/5699#issuecomment-348350792
 cd ${REPOINFRA_ROOT}
-TMP_GOPATH=$(mktemp -d)
 
-# no unit tests in vendor
-# previously we used godeps which did this, but `dep` does not handle this
-# properly yet. some of these tests don't build well. see:
-# ref: https://github.com/kubernetes/test-infra/pull/5411
-find ${REPOINFRA_ROOT}/vendor/ -name "*_test.go" -delete
+go install ./vendor/github.com/bazelbuild/bazel-gazelle/cmd/gazelle
```

Comment:
should we still use a tmp GOBIN instead of actually installing this?

Reply:
ditto for the rest of this commit

Reply:
Using a temp `GOBIN` seems to destroy caching. e.g. compare as currently written:
```console
$ for i in $(seq 5); do time ./verify/verify-bazel.sh ; done

real    0m1.115s
user    0m1.056s
sys     0m0.280s

real    0m0.218s
user    0m0.152s
sys     0m0.068s

real    0m0.195s
user    0m0.144s
sys     0m0.048s

real    0m0.192s
user    0m0.128s
sys     0m0.060s

real    0m0.207s
user    0m0.140s
sys     0m0.068s
```

with this diff:
```diff
$ git diff
diff --git a/verify/verify-bazel.sh b/verify/verify-bazel.sh
index b7bc2c2..2124390 100755
--- a/verify/verify-bazel.sh
+++ b/verify/verify-bazel.sh
@@ -20,24 +20,18 @@ set -o pipefail
 REPOINFRA_ROOT=$(git rev-parse --show-toplevel)
 # https://github.com/kubernetes/test-infra/issues/5699#issuecomment-348350792
 cd ${REPOINFRA_ROOT}
+TMP_GOBIN=$(mktemp -d)
 
-go install ./vendor/github.com/bazelbuild/bazel-gazelle/cmd/gazelle
-if ! which gazelle >/dev/null 2>&1; then
-    echo "Can't find gazelle - is your GOPATH's 'bin' in your PATH?" >/dev/stderr
-    echo "  GOPATH: ${GOPATH}" >/dev/stderr
-    echo "  PATH:   ${PATH}" >/dev/stderr
-    exit 1
-fi
-
-go install ./kazel
+GOBIN="${TMP_GOBIN}" go install ./vendor/github.com/bazelbuild/bazel-gazelle/cmd/gazelle
+GOBIN="${TMP_GOBIN}" go install ./kazel
 
 touch "${REPOINFRA_ROOT}/vendor/BUILD.bazel"
 
-gazelle_diff=$(gazelle fix \
+gazelle_diff=$("${TMP_GOBIN}/gazelle" fix \
   -external=vendored \
   -mode=diff)
 
-kazel_diff=$(kazel \
+kazel_diff=$("${TMP_GOBIN}/kazel" \
   -dry-run \
   -print-diff)
```
```console
$ for i in $(seq 5); do time ./verify/verify-bazel.sh ; done

real    0m1.485s
user    0m1.276s
sys     0m0.476s

real    0m1.188s
user    0m1.112s
sys     0m0.324s

real    0m1.302s
user    0m1.208s
sys     0m0.360s

real    0m1.355s
user    0m1.328s
sys     0m0.332s

real    0m1.282s
user    0m1.212s
sys     0m0.336s
```

Reply:
of course, it's also possible for the `gazelle` or `kazel` in your `$PATH` to not match the one you just installed.
we can use `$GOPATH/bin`, but that only works if you have a single GOPATH.

everything is terrible.

Reply:
(fixable, but terrible.)